fluent-merge/fluent-langneg.js

```diff
@@ -0,0 +1,76 @@
+(function (global, factory) {
+	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('fluent-syntax')) :
+	typeof define === 'function' && define.amd ? define('fluent-langneg', ['exports', 'fluent-syntax'], factory) :
+	(factory((global.FluentLangNeg = global.FluentLangNeg || {}),global.fluentSyntax));
+}(this, (function (exports,fluentSyntax) { 'use strict';
+
+/*
+ * Merge text files.
+ */
+function mergeFiles(...files) {
+  const resources = files.map(fluentSyntax.parse);
+  const merged = resources.reduce(merge);
+  return fluentSyntax.serialize(merged);
+}
+
+/*
+ * Merge AST trees into one tree.
+ */
+function mergeResources(...resources) {
+  const merged = resources.reduce(merge);
+  return merged;
+}
+
+/*
+ * Merge messages of two resources.
+ *
+ * The first resource is treated as the reference.
+ */
+function merge(res1, res2) {
+  const body = [];
+
+  for (const message of messages(res1)) {
+    const other = findMessage(res2, message.id.name);
+    if (other) {
+      body.push(other);
+    } else {
+      body.push(message);
+    }
+  }
+
+  const merged = new fluentSyntax.Resource(body, res1.comment);
+  merged.source = fluentSyntax.serialize(merged);
+
+  return merged;
+}
+
+/*
+ * Find a message in `res` by its `id`.
+ */
+function findMessage(res, id) {
+  for (const other of messages(res)) {
+    if (other.id.name === id) {
+      return other;
+    }
+  }
+
+  return null;
+}
+
+/*
+ * Create an iterator over Resource's messages.
+ */
+function* messages(res) {
+  for (const entry of res.body) {
+    if (entry.type === 'Message') {
+      yield entry;
+    }
+  }
+}
+
+exports.mergeFiles = mergeFiles;
+exports.mergeResources = mergeResources;
+
+Object.defineProperty(exports, '__esModule', { value: true });
+
+})));
```

Comment:
It's very confusing :) I have no idea what it's doing here, I guess I accidentally committed it. Thanks for spotting it!